ui/src/components/editor/shapeComponents/setDefinition/QuickQueryDialog.vue

```diff
@@ -0,0 +1,247 @@
+    if (isObjectHasKeys(entity, [IM.DEFINITION])) request.query = JSON.parse(entity[IM.DEFINITION]);
+  } else if (props.query) {
+    request.query = JSON.parse(props.query);
+  }
+
+  return request;
+}
+
+function addArguments(queryRequest: QueryRequest) {
+  queryRequest.argument = [];
+  for (const param of params.value) {
+    if ("Text search" === param.name) {
+      if (param.value) queryRequest.textSearch = param.value;
+    } else {
+      const argument = {
+        parameter: param.name
+      } as Argument;
+
+      if (isArrayHasLength(param.value)) {
+        argument.valueIriList = (param.value as []).map((summary: ConceptSummary) => {
+          return { "@id": summary.iri, name: summary.name } as TTIriRef;
+        });
+      } else {
+        argument.valueIri = { "@id": param.value.iri, name: param.value.name } as TTIriRef;
+      }
+      queryRequest.argument.push(argument);
+    }
+  }
+}
+
+async function search(searchTerm: any) {
+  if (!isObject(controller.value)) {
+    controller.value.abort();
+  }
+  controller.value = new AbortController();
+  suggestions.value = await EntityService.simpleSearch(searchTerm.query, filterDefaults.value, controller.value);
+}
+
+function debounceForSearch(searchTerm: any): void {
+  clearTimeout(debounce.value);
+  debounce.value = window.setTimeout(() => {
+    search(searchTerm);
+  }, 600);
+}
+
+async function testQuery() {
+  queryLoading.value = true;
+  try {
+    const queryRequest = await getQueryRequest();
+    addArguments(queryRequest);
+    queryResults.value = await QueryService.queryIM(queryRequest);
+    if (isObjectHasKeys(queryResults.value, ["entities"]) && isArrayHasLength(queryResults.value.entities))
+      cols.value = getCols(queryResults.value.entities[0]);
+    if (cols.value.some(col => col.field !== RDFS.LABEL)) {
+      await addNamesToResults(queryResults.value.entities);
+      cols.value.push({ field: "name" });
+    }
+  } catch (error: any) {
+    if (error?.response?.data) toast.add(new ToastOptions(ToastSeverity.ERROR, "An error occurred: " + error?.response?.data.debugMessage));
+  }
+
+  queryLoading.value = false;
+}
+
+async function addNamesToResults(entities: any[]) {
+  const results = await EntityService.getNames(entities.map(entity => entity["@id"]));
+  for (const namedEntity of results) {
+    const found = entities.find(entity => entity["@id"] === namedEntity["@id"]);
+    if (found) found.name = namedEntity.name;
+  }
+}
+
+function getCols(entity: any) {
+  const cols = [];
+  for (const key of Object.keys(entity)) {
+    cols.push({ field: key });
+  }
+  return cols;
+}
+
+function onDownload(): void {
+  if (isObjectHasKeys(queryResults.value, ["entities"]) && isArrayHasLength(queryResults.value.entities)) {
+    const heading = ["name", "iri"].join(",");
+    const body = queryResults.value.entities.map((row: any) => '"' + [row.name, row["@id"]].join('","') + '"').join("\n");
```

Comment:
why not use string concatenation, concatenation is faster and clearer

row.name + "," + row["@id"]